crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -390,13 +755,13 @@ pub struct ResState<T> {
     marker: PhantomData<T>,
 }
 
-impl<'a, T: Resource> SystemParam for Res<'a, T> {
+impl<'a, T: Resource> SystemParam<Optional> for Res<'a, T> {
     type State = ResState<T>;
 }
 
-// SAFETY: Res ComponentId and ArchetypeComponentId access is applied to SystemMeta. If this Res
+// SAFETY: Res ComponentI d and ArchetypeComponentId access is applied to SystemMeta. If this Res
```

Comment:
```suggestion
// SAFETY: Res ComponentId and ArchetypeComponentId access is applied to SystemMeta. If this Res
```

This change shouldn't exist and looks like a typo.